frontend/lib/ui/artefact_page/test_result_expandable.dart

```diff
@@ -73,24 +78,81 @@ class _TestResultOutputExpandable extends StatelessWidget {
   }
 }
 
-class PreviousTestResultsWidget extends StatelessWidget {
-  const PreviousTestResultsWidget({super.key, required this.previousResults});
+class _PreviousTestResultsWidget extends ConsumerWidget {
+  const _PreviousTestResultsWidget({
+    required this.currentResult,
+    required this.previousResults,
+  });
 
+  final TestResult currentResult;
   final List<PreviousTestResult> previousResults;
 
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, WidgetRef ref) {
+    final artefactId =
+        AppRoutes.artefactIdFromUri(AppRoutes.uriFromContext(context));
+    final currentVersion = ref
+            .watch(
+              artefactProvider(artefactId)
+                  .select((data) => data.whenData((a) => a.version)),
+            )
+            .value ??
+        '';
+
+    final statusGroups = {
+      currentVersion: [
+        PreviousTestResult(
+          artefactId: artefactId,
+          status: currentResult.status,
+          version: currentVersion,
+        ),
+      ],
+    };
+    for (final result in previousResults) {
+      final statuses = statusGroups[result.version];
+      if (statuses != null) {
+        statuses.add(result);
+      } else {
+        statusGroups[result.version] = [result];
+      }
+    }
+
     return Row(
-      children: previousResults
-          .map(
-            (e) => InkWell(
-              onTap: () => navigateToArtefactPage(context, e.artefactId),
+      children: statusGroups.entries
+          .mapIndexed<Widget>(
+            (groupIndex, entry) => InkWell(
```

Comment:
We seem to have quite a few elements inside this `InkWell` widget, perhaps we can move it to a separate private method that returns a `PreviousResultStatusGroup` widget